tests/TestExpectedFails.hs

```diff
@@ -26,7 +28,11 @@ main = defaultMain $
 
   , expectFail $ testCase "takes too long" $ threadDelay 2000000
 
-  , expectFail $ goldenVsString "hello" "hello.out" $ return $ error "not golden"
+  , expectFail $ goldenVsString "hello_golden_err" "hello_err.out" $
+      return $ error "not golden"
+
+  , expectFail $ goldenVsString "hello_golden" "hello.out" $
+      return "good bye!"
```

Comment:
There is a slight semantic difference between these two golden tests:
One fails during result evaluation whereas other produces a different result.
I have tested different invariants manually:
1. Golden file doesn't exist and is created for the first time;
2. Golden file exists, matches result and no `--accept` flag is specified;
3. Golden file exists, doesn't match and no `--accept` flag is specified;
3. Golden file exists and `--accept` flag is specified;
In all the cases I tried results matched my expectations.